https://issues.redhat.com/browse/ACM-22992 ClusterRoleBindings for ClusterPermissions (#8329)
open-cluster-management/rhacm-docs: secure_clusters/rbac_implement_rhacm.adoc

Complete the following step to create a `ClusterPermission` resource for creating multiple cluster role bindings:

. To create a `ClusterPermission` resource to have many cluster role bindings, run the following command:

+
[source,bash]
----
oc notsure what the rest mght be if there is a command
----
+
Your resource might resemble the following YAML with the specified `clusterRoleBindings` field:

+
[source,yaml]
----
apiVersion: rbac.open-cluster-management.io/v1alpha1
kind: ClusterPermission
metadata:
  name: clusterpermission-multiple-clusterrolebindings
spec:
  clusterRoleBindings:
    - name: multi-crb-binding1
      roleRef:
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
        name: argocd-application-controller-1
      subject:
        kind: User
        name: user1
    - name: multi-crb-binding2
      roleRef:
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
        name: argocd-application-controller-3
      subjects:
        - kind: User
          name: user2
        - kind: Group
          name: group1
----

> 

Reviewer: Introduce MCE RBAC

Author: The link is mentioned on line 6, should I still add the link here?

Reviewer: ok, no that was hidden and I didn't see it here. That is good!

View the following sections for more information on RBAC for specific lifecycles for {acm-short}:

* <<cluster-mgmnt-RBAC,Cluster management RBAC>>
* <<application-lifecycle-RBAC,Application lifecycle RBAC>>
* <<governance-lifecycle-RBAC,Governance lifecycle RBAC>>
* <<observability-RBAC,Observability RBAC>>

[#cluster-mgmnt-RBAC]
== Cluster management RBAC

> To perform cluster management actions, you need access to your managed and hub cluster. If you want to create multiple cluster role bindings, you can use the `clusterRoleBindings` field to create multiple cluster role bindings in a single `ClusterPermission` resource.

Reviewer: ```suggestion
For cluster management actions, you need access to your managed cluster and your hub cluster. If you want to create multiple cluster role bindings, you can use the `clusterRoleBindings` field to create multiple cluster role bindings in a single `ClusterPermission` resource.
```

Reviewer: Does this get added to the new features doc?

Author: yup! thanks for the reminder!

Complete the following step to create a `ClusterPermission` resource for creating multiple cluster role bindings:

> . To create a `ClusterPermission` resource to have many cluster role bindings, run the following command:

Reviewer: You have a task here, but it is inside a concept.

Reviewer: Concept: Title does not begin with a verb, has not steps.
Task: Title begins with a verb, has a procedure.

Reviewer: These either need different headers, or no task information in them. 

Example (I would make this change to all and simplify)

_Enabling RBAC for cluster management_

+
[source,bash]
----

> oc notsure what the rest mght be if there is a command

Reviewer: We need that info on line 27, @fxiang1 , I don't think this should be approved?

Reviewer 2: Good catch Brandi! Right, this line should not be here. Please remove. Thanks!

Reviewer 2: I guess we need a command to replace it. You can use this:
```
oc create clusterpermission clusterpermission-multiple-clusterrolebindings -n <cluster-name>
```

----
+
Your resource might resemble the following YAML with the specified `clusterRoleBindings` field:

+
[source,yaml]
----
apiVersion: rbac.open-cluster-management.io/v1alpha1
kind: ClusterPermission
metadata:
  name: clusterpermission-multiple-clusterrolebindings
spec:
  clusterRoleBindings:
    - name: multi-crb-binding1
      roleRef:
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
        name: argocd-application-controller-1
      subject:
        kind: User
        name: user1
    - name: multi-crb-binding2
      roleRef:
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
        name: argocd-application-controller-3
      subjects:
        - kind: User
          name: user2
        - kind: Group
          name: group1
----

[#application-lifecycle-RBAC]

> == Application lifecycle RBAC

Reviewer: same with the header here.

> When you create an application, the `_subscription_` namespace is created and the configuration map is created in the `_subscription_` namespace. You must also have access to the `_channel_` namespace. When you want to apply a subscription, you must be a subscription administrator. For more information about managing applications, see link:../applications/allow_deny.adoc#creating-allow-deny-list[Creating an allow and deny list as subscription administrator]. 

Reviewer: _When you create an application, the `_subscription_` namespace is created and the configuration map is created in the `_subscription_` namespace._

this was confusing, it seems like it supposed to be the first thing happens, then the second thing? The way it is written, it seemed like everything happened at once, which would not make sense.

When you create an application, the `_subscription_` namespace is created,  **then** the configuration map is created **within** the `_subscription_` namespace. **(????)**

View the following application lifecycle RBAC operations:

- Create and administer applications on all managed clusters with a user named `username`. You must create a cluster role binding and bind it to `username`. Run the following command:

+
[source,bash]
----
oc create clusterrolebinding <role-binding-name> --clusterrole=open-cluster-management:cluster-manager-admin --user=<username>
----
+

> This role is a superuser, which has access to all resources and actions. You can create the namespace for the application and all application resources in the namespace with this role.

Reviewer: ```suggestion
This role assigns superuser access to all resources and actions. As a superuser, you can create the namespace for the application and all application resources in the namespace with this role.
```

Reviewer: You switch voice in the middle of this for some reason, need to stick with one or the other. Hopefully this works, but if not feel free to change it.

(The "role" cannot be a superuser. The user in that role is the superuser.)

+
[source,bash]
----
oc create clusterrolebinding <role-binding-name> --clusterrole=open-cluster-management:subscription-admin --user=<username>
----

- To view an application on a managed cluster named `cluster-name` with the user named `username`, create a cluster role binding to the `open-cluster-management:view:` cluster role and bind it to `username`. Enter the following command:

+
[source,bash]
----
oc create clusterrolebinding <role-binding-name> --clusterrole=open-cluster-management:view:<cluster-name> --user=<username>
----
+
This role has read access to all `application` resources on the managed cluster, `cluster-name`. Repeat this if access for other managed clusters is required.

- Create a namespace role binding to the `application` namespace using the `view` role and bind it to `username`. Enter the following command:

+
[source,bash]
----
oc create rolebinding <role-binding-name> -n <application-namespace> --clusterrole=view --user=<username>
----
+
This role has read access to all `application` resources in the `application` namspace. Repeat this if access for other applications is required.

[#governance-lifecycle-RBAC]
== Governance RBAC

> To perform Governance operations, you need access to the namespace where the policy is created, along with access to the managed cluster where the policy is applied. The managed cluster must also be part of a `ManagedClusterSet` that is bound to the namespace. To continue to learn about `ManagedClusterSet`, see link:../clusters/cluster_lifecycle/clusterset_intro.adoc#managedclustersets-intro[ManagedClusterSets Introduction].

Reviewer: looks good, same with the header since there are procedures here.

+
[source,bash]
----
oc create clusterrolebinding <role-binding-name> --clusterrole=open-cluster-management:subscription-admin --user=<username>
----

- To view an application on a managed cluster named `cluster-name` with the user named `username`, create a cluster role binding to the `open-cluster-management:view:` cluster role and bind it to `username`. Enter the following command:

+
[source,bash]
----
oc create clusterrolebinding <role-binding-name> --clusterrole=open-cluster-management:view:<cluster-name> --user=<username>
----
+
This role has read access to all `application` resources on the managed cluster, `cluster-name`. Repeat this if access for other managed clusters is required.

- Create a namespace role binding to the `application` namespace using the `view` role and bind it to `username`. Enter the following command:

+
[source,bash]
----
oc create rolebinding <role-binding-name> -n <application-namespace> --clusterrole=view --user=<username>
----
+
This role has read access to all `application` resources in the `application` namspace. Repeat this if access for other applications is required.

[#governance-lifecycle-RBAC]
== Enabling RBAC for Governance

> To perform Governance operations, you need access to the namespace where the policy is created, along with access to the managed cluster where the policy is applied. The managed cluster must also be part of a `ManagedClusterSet` that is bound to the namespace. To continue to learn about `ManagedClusterSet`, see link:../clusters/cluster_lifecycle/clusterset_intro.adoc#managedclustersets-intro[ManagedClusterSets Introduction].

Author: ```suggestion
For Governance actions, you need access to the namespace where the policy is created, along with access to the managed cluster where the policy is applied. The managed cluster must also be part of a `ManagedClusterSet` that is bound to the namespace. To continue to learn about `ManagedClusterSet`, see link:../clusters/cluster_lifecycle/clusterset_intro.adoc#managedclustersets-intro[ManagedClusterSets Introduction].
```
